src/foam/util/AddressUtil.java

```diff
@@ -0,0 +1,41 @@
+package foam.util;
+
+import java.util.regex.Pattern;
+
+public class AddressUtil {
+
+  public static String[] parseAddress(String address1, String address2) {
+    var suitePattern = Pattern.compile("/\\d+/g");
```

Comment:
Patterns are expensive to compile, so move this to a constant, like: https://github.com/foam-framework/foam2/blob/3a5cb639d8e01dfabe04c50f2266b54b4511854b/src/foam/util/Email.java#L13